timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -112,11 +156,11 @@ def freq(self):
         return freq
 
     def iter_items(self) -> Iterable[Any]:
-        return iter(self.index.levels[0])
+        return iter(self._item_index)
```

Comment:
This also fixes a bug that we currently have on `master`, where `iter_items` is broken for subsets of a `TimeSeriesDataFrame` selected via `loc` :+1: